From 328ab523f2b45e7d3e712cb7707a70dca5005234 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 18 Feb 2025 19:29:23 +0100
Subject: [PATCH 1537/1695] plat-stm32mp2: stm32mp25_pwr: remove fixme in
 driver

Remove a fixme in PWR regulator driver as it is NOT a fix
but a choice: use static pre-alloc instead dynamic allocation
as it is possible in new regulator framework.

The current choice is aligned with stm32mp1_pwr.c to avoid small
allocation during probe and because the number of PWR regulator is fixed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 39f40e4d9654e32daf3e6dfc97833610cd267b17 plat-stm32mp2: stm32mp25_pwr: remove fixme in driver]
Fixes: d4bb9f98f4af ("plat-stm32mp2: Add PWR regulators driver")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ieb83eec179cef827a3a030100980c458ce9f791f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/438196
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
index c26003a0c..d815d2ff1 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
@@ -532,7 +532,7 @@ static struct pwr_regu pwr_regulators[PWR_REGU_COUNT] = {
 		.regulator = pwr_regu_device + (_id),	\
 	}
 
-/* (FIXME: not needed) Preallocated regulator devices */
+/* Preallocated regulator devices */
 static struct regulator pwr_regu_device[PWR_REGU_COUNT];
 
 /* Not const to allow probe to reassign ops if device is a fixed_iod */
-- 
2.47.2

